src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -207,88 +207,74 @@ void MVN::initSupportedPrimitiveDescriptors() {
+    const std::vector<MemoryDescArgs> nodeDescriptorsList = factory->getProperMemoryDescriptors(descs);
+
+    // Build supported primitive descriptors; prefer nspc when formats are enforced
+    std::vector<NodeDesc> nspc_spds;
+    std::vector<NodeDesc> other_spds;
+    for (const auto& nodeDescriptors : nodeDescriptorsList) {
+        NodeConfig nodeConfig;
+        nodeConfig.inConfs.resize(getParentEdges().size());
 
-    // TODO [DS]: inplace
-    bool canBeInplace = !isDynamicNode() && (inputPrecision.size() == outputPrecision.size()) &&
-                        (getParentEdgeAt(0)->getParent()->getChildEdges().size() == 1) &&
-                        !getParentEdgeAt(0)->getParent()->isConstant();
+        const auto& outDesc = nodeDescriptors.at(ARG_DST);
+        const auto outPrecision = outDesc->getPrecision();
 
-    const size_t inputsNum = getParentEdges().size();
+        // Input config
+        const int inPlace = (!isDynamicNode() && inputPrecision.size() == outPrecision.size() &&
+                             getParentEdgeAt(0)->getParent()->getChildEdges().size() == 1 &&
+                             !getParentEdgeAt(0)->getParent()->isConstant())
+                                ? 0
+                                : -1;
 
-    // Create supported primitive descriptors for each layout configuration
-    for (const auto& config : configurations) {
-        NodeConfig nodeConfig;
-        nodeConfig.inConfs.resize(inputsNum);
-        nodeConfig.outConfs.resize(1);
-        nodeConfig.inConfs[0].constant(false);
-        nodeConfig.outConfs[0].constant(false);
-        nodeConfig.inConfs[0].inPlace(-1);
-        nodeConfig.outConfs[0].inPlace(canBeInplace ? 0 : -1);
-        if (inputsNum == 2) {
+        nodeConfig.inConfs[0] = PortConfig(nodeDescriptors.at(ARG_SRC_0), BlockedMemoryDesc::SKIP_OFFSET_MASK, -1);
+        nodeConfig.outConfs.emplace_back(outDesc, BlockedMemoryDesc::SKIP_OFFSET_MASK, inPlace);
+
+        // Axes input if present
+        if (getParentEdges().size() == 2) {
             nodeConfig.inConfs[1].setMemDesc(
                 std::make_shared<CpuBlockedMemoryDesc>(ov::element::i32, getInputShapeAtPort(1)));
             nodeConfig.inConfs[1].constant(true);
         }
 
-        // Use the layout-specific descriptors
-        nodeConfig.inConfs[0].setMemDesc(config.first);
-        nodeConfig.outConfs[0].setMemDesc(config.second);
+        const bool is_nspc = nodeDescriptors.at(ARG_SRC_0)->hasLayoutType(LayoutType::nspc);
+        if (is_nspc) {
+            nspc_spds.emplace_back(nodeConfig, ov::intel_cpu::impl_desc_type::undef);
+        } else {
+            other_spds.emplace_back(nodeConfig, ov::intel_cpu::impl_desc_type::undef);
+        }
+    }
 
-        supportedPrimitiveDescriptors.emplace_back(nodeConfig, ov::intel_cpu::impl_desc_type::undef);
+    if (enforce_formats) {
+        // If formats enforced, and nspc variants exist, keep only nspc to avoid accidental planar selection
```

Comment:
I didn't get this logic.
Maybe I am wrong, but if memory format is enforced then 'support' method of the executor implementation will return true only for the corresponding memory format and we will have either 'ncsp' or 'nspc' executors, but not both at the same time.
But maybe there is some tricky logic in MVN node I am not aware about.